From f17725108809efaaf139f141808d537d14db3708 Mon Sep 17 00:00:00 2001
From: Junwei Zhang <Jerry.Zhang@amd.com>
Date: Thu, 30 Aug 2018 17:35:40 +0800
Subject: [PATCH 3/3] amdgpu: add error return value for finding bo by cpu
 mapping (v2)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If nothing is found, error should be returned.

v2: udpate the error value different from parameter check

Signed-off-by: Junwei Zhang <Jerry.Zhang@amd.com>
Reviewed-by: Christian König <christian.koenig@amd.com>
Reviewed-by: Michel Dänzer <michel.daenzer@amd.com>
Signed-off-by: Laurent Carlier <lordheavym@gmail.com>
---
 amdgpu/amdgpu_bo.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/amdgpu/amdgpu_bo.c b/amdgpu/amdgpu_bo.c
index 2f4f90f6..a2fc5250 100644
--- a/amdgpu/amdgpu_bo.c
+++ b/amdgpu/amdgpu_bo.c
@@ -549,8 +549,9 @@ int amdgpu_find_bo_by_cpu_mapping(amdgpu_device_handle dev,
 				  amdgpu_bo_handle *buf_handle,
 				  uint64_t *offset_in_bo)
 {
-	uint32_t i;
 	struct amdgpu_bo *bo;
+	uint32_t i;
+	int r = 0;
 
 	if (cpu == NULL || size == 0)
 		return -EINVAL;
@@ -577,10 +578,11 @@ int amdgpu_find_bo_by_cpu_mapping(amdgpu_device_handle dev,
 	} else {
 		*buf_handle = NULL;
 		*offset_in_bo = 0;
+		r = -ENXIO;
 	}
 	pthread_mutex_unlock(&dev->bo_table_mutex);
 
-	return 0;
+	return r;
 }
 
 int amdgpu_create_bo_from_user_mem(amdgpu_device_handle dev,
-- 
2.18.0

